### PR TITLE
[CIS-792] Make models thread safe

### PR DESCRIPTION
**Pull request by VojtaStavik**
There was a regression introduced by #906 which made our models thread-unsafe. This PR fixes the problem. Instead of the `@Cached` wrapper, we're using `@CoreDataLazy` which evaluates the values on the correct `NSManagedObjectContext`.

### CODE REVIEW

File: Sources/StreamChat/Utils/CoreDataLazy.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            let perform = {
+                log.assert(self.computeValue != nil, "You must set the `computeValue` closure before accessing the cached value.")
+                returnValue = self.computeValue()
+            }
+            
+            if let context = context {
+                context.performAndWait { perform() }
+            } else {
+                // This is really a fallback for cases like tests, mocks, and edge cases. Even though `NSManagedObject.managedObjectContext`
+                // is optional, we probably won't reach this point during a normal production run.
```
Comment (olejnjak) [inline Sources/StreamChat/Utils/CoreDataLazy.swift:41]: If we consider this to be unreachable in production, it might be better to add some assertion?

**Review by olejnjak — APPROVED** [review]

File: Sources/StreamChat/Utils/CoreDataLazy.swift (hunk 1 above)
Comment (VojtaStavik) [reply Sources/StreamChat/Utils/CoreDataLazy.swift:41]: @olejnjak I didn't say it's "unreachable" I said "we probably won't reach" 😅 The comment is actually outdated. MessageDTO uses this for the cases where the results are empty and we don't need to use the context queue. I'll update it.
